### DISCUSSION TITLE
Inverter + E/I showing 0 values for Power and Current A and B

### DISCUSSION THREAD

**Discussion by aarondrabeck** [WillCodeForCats/solaredge-modbus-multi#298] [General]
I am seeing 0 values for Power A and B.   I am expecting to see the values here for Line 1 and Line 2.  Any idea what could be going on?

![image](https://user-images.githubusercontent.com/9825025/233498003-8171eeca-1722-4ccb-97ea-ddff95d8fb49.png)

**Comment by WillCodeForCats** [comment]
Only one of the two functions of a dual function meter will be exposed to modbus. You can see it's the production function because it says "hardware: production" in device info.

This is not a bug: this is how SolarEdge has chosen implement a dual function meter over modbus.

**Reply by aarondrabeck** [reply]
Is there a way to have it send the I/E side?  Otherwise I get the Production from the Inverter both on the meter and from the invertor device which isn't enough info to tell anything about consumption or export

**Reply by WillCodeForCats** [reply]
No, as far as I know there is no option to change which meter function it reports over modbus. Ideally it would report both functions as two separate modbus meters, but it doesn't and only SolarEdge can change that in firmware.

You would have to reconfigure the meter as an I/E meter only and then go inside the inverter to psychically disconnect the built-in production CT from the L3 CT input input. I don't know if this requires also removing the production CT, which would also be required if it doesn't have a built in burden resistor (current transformers can be dangerous when operated without a connected load so some have built-in resistors to make them safe).

If you have a utility program (SRECs, performance incentives, etc.), PPA or lease with production guarantees that require a revenue grade production meter then you shouldn't disconnect it. In that case the only option is to get a second meter dedicated for I/E metering.

**Reply by WillCodeForCats** [reply]
You could also try reporting it as an issue to SolarEdge support or your installer but I don't know if that would get any results.

Basically it needs to internally split and report it as two separate meters over modbus: one Production meter and one Export+Import meter.

**Reply by aarondrabeck** [reply]
Thanks for the suggestions!

The SetApp is detecting the dual meter somehow and only allows 2 options.  Invertor or Invertor + import/export.  I could set to invertor and move one of the two L1 or L2 over to #3 I suppose, but I would need both L1 and L2 data to make it useful.  

The invertor seems to have a separate way to measure power, because I can remove the whole meter from the config and It still reports output.  Also, the L3 production value and invertors production meter are different values, which probably shouldn't be the case...have not figured that out yet.  

Anyway, for now I will use a z-wave power meter for home consumption and I can calculate the rest from there.  Hopefully SolarEdge can report both values in the future.

**Reply by WillCodeForCats** [reply]
Correct the inverter can show production without the meter, but the meter is a "revenue grade" option. It's more accurate. The inverter can be ordered with or without a revenue grade meter built in. If it's ordered with it comes with a special internal CT wired to the L3 CT input for the higher accuracy production metering.

**Reply by ixs** [reply]
For what's it worth, current SolarEdge systems seem to support both Input and Output values via modbus.
Tested with both the _Inline Meter_ `MTR-240-3PC1-D-A-MW` as well as the _Connection Meter with Modbus Connection_ `SE-MTR-3Y-400V-A 100A`.

Home Assist sensors: 
<img width="308" alt="modbus output in home assist" src="https://github.com/user-attachments/assets/fd6f6da7-c627-44c1-8725-403bf2851bc3">

SetApp configuration view: 
<img width="400" alt="meter configuration in setapp" src="https://github.com/user-attachments/assets/8b5a9ba1-a09e-4a10-a9d9-977d48bbf1f3">

I think we can probably consider this topic resolved successfully.

**Reply by KevinCathcart** [reply]
At this point it has been a fair while now that the North American P+I/E dual-function meters (SE-RGMTR-1D-240C-B and -C) have been getting reported as two meters over external (SunSpec) modbus, one being "Hardware: Production" the other being "Hardware: Export+Import". (This does cause a spurious warning from the integration about a duplicate serial number, since both give the same serial, as they represent the same physical meter. It might be reasonable to suppress that warning for SE-RGMTR-1D-240C-* models.)

If users are not seeing both, I'd recommend they try updating their inverter firmware.

I'm not entirely sure what happens if they have a second meter installed. I'm not even sure if system normally tries to map the SunSpec meter slots to the meter's ID (1,2, or 3) or if it just puts each meter in the first unused SunSpec meter slot. If it does normally try to preserve the IDs, then users with multiple meters one of which is dual function might need their second meter set a specific id to keep it from overwriting the Export/Import virtual meter.

I'm not personally aware of any other meters that are truly dual-function like those. I suppose a 480 split phase P+I/E meter could be made, or a variety of dual function meters would be possible for places that do single non-split 240v. If such a meter does exist it might not show up as two like the North American P+I/E meters, but at least for those specific meters, I'm pretty sure this is just resolved now.

**Reply by WillCodeForCats** [reply]
The inverter just maps meters to slots 1, 2, and 3 in the order they are discovered as far as I'm aware (possibly address order).

**Reply by Tcoton** [reply]
Which firmware version is it supposed to work with? 
I have a SE-RGMTR-1D-240C-C with inverter firmware 4.23.36 and I cannot see the export+import meter. My system is brand new, installed this year and I have very limited control over it because of the contract I have with the installer. I requested a firmware update, not sure they will proceed.

**Reply by KevinCathcart** [reply]
hmm... I currently have 4.20.36, on a SE5000H-US000BNI4, with my SE-RGMTR-1D-240C-B. I find it hard to believe that they would only support meter split out for the -B and not the -C, since they are such similar meters. I assume your meter is configured in the commissioning UI as "Inverter Production + E/I"? 

I wonder if they broke it in newer firmware? I'm a little reluctant to upgrade to the newer firmware to test simply because they make it such a pain to downgrade firmware.

**Reply by Tcoton** [reply]
Yes my meter is configured as "Inverter Production + E/I" and I have searched for 3 days straight why I couldn't get the data through Modbus TCP, I even pulled a CAT6 cable to connect to it Modbus. So either there is something in the firmware blocking it or the registers are specific and cannot be auto detected. I can't find a command line tool for Linux to scan the registers of this thing (yet).

**Reply by WillCodeForCats** [reply]
Turn on debug and reload the integration. It will show what was returned for all three meter slots in debug log messages when the integration start up.

**Reply by Tcoton** [reply]
Could not spot any oddities in the debug log file, no second meter.
[home-assistant_solaredge_modbus_multi_2025-11-05T13-24-49.622Z.log](https://github.com/user-attachments/files/23361302/home-assistant_solaredge_modbus_multi_2025-11-05T13-24-49.622Z.log)

**Reply by Tcoton** [reply]
I got it working, for some reason the installers never configured the meter properly. I went again at it today after reading some more Solaredge documentation about meter installation. I double checked the CT rating of the CT installed in my electrical panel and set this value in the Inverter's commissioning  portal.  I instantly got the export-import data info in the portal and in the app and after reloading the integration, meter 2 appeared! The documentation says that if CT ratings are set to 0, there will be no data transfer.

Site Communication --> RS-485-1 (Modbus Multi-Device)--> Meter 1 --> Export+Import CT rating => set the rating of the CT installed.

<img width="295" height="639" alt="IMG_9143" src="https://github.com/user-attachments/assets/d416e182-b78b-4de7-9b15-ceadd78693d4" />

Now, I have to redo my dashboards to get the proper power flow in the Tesla look alike thingy.

**Reply by KevinCathcart** [reply]
Yeah that makes sense. The current clamps are literally documented as optional for this meter, as its primary purpose is the Revenue grade metering for SRECs. The other features like expor limitation, I/E monitoring and BUSBar PCS are all secondary. But it is wild that an installer would install the current clamps and wire them up, but forget to input them in the meter setup. Glad to hear that it does all work once you fixed that.
